presto-jdbc/src/main/java/io/prestosql/jdbc/ConnectionProperties.java

```diff
@@ -34,13 +34,49 @@
 import static io.prestosql.client.ClientSelectedRole.Type.ALL;
 import static io.prestosql.client.ClientSelectedRole.Type.NONE;
 import static io.prestosql.jdbc.AbstractConnectionProperty.checkedPredicate;
+import static io.prestosql.jdbc.ConnectionProperties.Predicates.IF_KEY_STORE;
+import static io.prestosql.jdbc.ConnectionProperties.Predicates.IF_SSL_ENABLED;
+import static io.prestosql.jdbc.ConnectionProperties.Predicates.IF_SSL_VERIFICATION_ALLOWED;
+import static io.prestosql.jdbc.ConnectionProperties.Predicates.IF_SSL_VERIFICATION_ENABLED;
+import static io.prestosql.jdbc.ConnectionProperties.Predicates.IF_TRUST_STORE;
+import static io.prestosql.jdbc.ConnectionProperties.Predicates.NO_HTTP_PROXY;
+import static io.prestosql.jdbc.ConnectionProperties.Predicates.NO_SOCKS_PROXY;
 import static java.util.Collections.unmodifiableMap;
 import static java.util.Objects.requireNonNull;
 import static java.util.function.Function.identity;
 import static java.util.stream.Collectors.toMap;
 
 final class ConnectionProperties
 {
+    enum SslVerificationMode
+    {
+        FULL, CA, NONE;
+    }
+
+    static class Predicates
+    {
+        static final Predicate<Properties> IF_KEY_STORE =
+                checkedPredicate(properties -> SSL_KEY_STORE_PATH.getValue(properties).isPresent());
+
+        static final Predicate<Properties> IF_SSL_ENABLED =
+                checkedPredicate(properties -> SSL.getValue(properties).orElse(false));
+
+        static final Predicate<Properties> IF_SSL_VERIFICATION_ALLOWED =
+                checkedPredicate(properties -> SSL.isProvided() && SSL.getValue(properties).get() || !SSL_VERIFICATION.isProvided());
```

Comment:
```suggestion
                checkedPredicate(properties -> SSL.getValue(properties).isPresent() && SSL.getValue(properties).get() || !SSL_VERIFICATION.getValue(properties).isPresent());
```

No need for `isProvided`

Reply:
Since we set a default value to `SSL_VERIFICATION`, `SSL_VERIFICATION.getValue(properties).isPresent()` is always true.

Therefore, if we use `SSL_VERIFICATION.getValue(properties).isPresent()` in the Predicate, as long as the `SSL` property is not set, the following exception will be thrown:
`java.sql.SQLException: Connection property 'SSLVerification' is not allowed`

Reply:
I don't get it. How `getValue(properties)` return a default value when there is no value? If that would be a case then we don't need `getValueOrDefault`.

Reply:
In `PrestoDriverUri` the default properties are merge into a `Properties` object which will be validated.
As a result, the key of `SSL_VERIFICATION` will always exist in the `Properties` object even if it is not provided in the uri or driver properties.

Therefore, when the `SSL` property is not set, the exception `java.sql.SQLException: Connection property 'SSLVerification' is not allowed` will be thrown.

Reply:
Setting a default value to the `SSL` property should solve issue. I actually once tried this approach but failed to make the program function correctly. Then I used to `isProvided` approach.

But now I think that the "setting default value to SSL" approach did not work maybe was just because I set a wrong condition to check if the `SSL_VERIFICATION` is allowed in the `Predicate`.

I will try to resolve this problem by setting a default value to the `SSL` property and avoid using the `isProvided` variable.
I apologize for committing the code without thinking about this problem thoroughly enough.